src/builder.rs

```diff
@@ -184,9 +187,12 @@ impl<C> ClientBuilder<C> {
             ServiceAccount(service_config) => Some(
                 create_service_auth(
                     match service_config {
-                        ServiceAccountAuth::Path(path) => path.into_os_string(),
-                        ServiceAccountAuth::EnvVar => std::env::var_os(SERVICE_ACCOUNT_ENV_VAR)
-                            .ok_or(CreateBuilderError::CredentialsVarMissing)?,
+                        ServiceAccountAuth::Path(path) => Some(path.into_os_string()),
+                        ServiceAccountAuth::EnvVar => Some(
+                            std::env::var_os(SERVICE_ACCOUNT_ENV_VAR)
+                                .ok_or(CreateBuilderError::CredentialsVarMissing)?,
+                        ),
+                        _ => None,
```

Comment:
As a convention I prefer using exhaustive matches rather than catch-all `_`, just in case the enum changes in the future

Reply:
@rnarubin I replaced the `_` with an exhaustive match, good call.